client/ratings_reviews/components/ReviewFormModal.jsx

```diff
@@ -61,80 +81,120 @@ class ReviewFormModal extends React.Component {
         name: '',
         email: '',
         photos: [],
-        characteristics: {}
+        characteristics: {},
+        photoCount: 0,
+        show: false,
+        showUploadPhotosButton: true
       }, this.props.closeModal);
     }).catch((error) => {
       console.log(error);
     });
   }
 
+  toggleModal() {
+    this.setState({
+      show: !this.state.show
+    });
+  }
+
+  validateForm() {
+    let valid = true;
+    let properties = ['rating', 'summary', 'body', 'name', 'email', 'characteristics'];
+    properties.forEach(property => {
+      let invalidCharacteristics = (property === 'characteristics' && Object.keys(this.state[property]).length !== Object.keys(this.props.meta.characteristics).length);
+      let invalidEmail = (property === 'email' && this.state[property].indexOf('@') === -1);
```

Comment:
I had to validate email addresses in my widget, too. It's hard to be super-thorough about it, but you might want to consider more than just the existence of an @ symbol.

Reply:
Hi @matthewwrobel, you're absolutely right. I will look into ways to detect emails. My first thought is to use regex to detect [string]@[string].[string] pattern but I'm not sure if that would be enough. Will look into it!